### DISCUSSION TITLE
Printing issue with Single guest session over NFS

### DISCUSSION THREAD

**Discussion by wolfclan68** [ltsp/ltsp#207] [Q&A]
I have the single guest session over NFS running perfectly, with one minor item. I was having issues with all 30+ of my network printers automatically installing so I set the remote and local browse function in cups-browsed.conf to none, but still had the issue so I disabled the avahi-daemon. I then manually added the few printers I wanted to be available which worked just fine in the guest user account. However, when the netboot client comes up it shows no printers installed. Any ideas?

I do have a workaround since we have a PaperCut server, I added the MobilityPrint extension to the Chrome browser and they can enter their credentials to submit the job. It works, but I was trying to make it as simple as possible for the users, especially since they are elementary students.

Thank you for your time as always.

**Comment by alkisg** [comment]
See https://github.com/ltsp/ltsp/issues/8#issuecomment-554458466

**Comment by wolfclan68** [comment]
I've followed what you mentioned in the referenced issue. I've run the sudo cupsctl _share_printers=1 from the admin account, the printers show shared in localhost:631 inside the guest account on the server, then added the line to the ltsp.conf.  My added line is this, maybe a typo: **echo "192.168.17.39 $SERVER" > /etc/cups/printers.conf**. I changed the client.conf to printers.conf because of no client.conf existing there and the printers.conf seemed to include all the printer data. But no printers show in the netboot client. However, they do appear in settings and I can add them to the client, but obviously disappear on reboot.

**Comment by alkisg** [comment]
Use the **exact** line as mentioned there. Run `man client.conf` to read about what "ServerName" is. $SERVER **is** your server IP, do not replace it with anything.

After the client boots, if you run `cat /etc/cups/client.conf`, you should see the server IP there. And cups should connect to the server and list the server printers.

**Comment by alkisg** [comment]
I updated https://github.com/ltsp/ltsp/issues/8#issuecomment-554458466 and I tested it and it works fine for me, give it another go.

**Comment by wolfclan68** [comment]
Worked perfectly! This whole config with the guest setup and now printing working is quite elegant. Thank you very much for the time you have put into this and patiently answering questions for us Linux noobs.

**Comment by alkisg** [comment]
Great, I will push this printing solution as the upstream default when I get some free time.
